src/PyProxyHandler.cc

```diff
@@ -24,16 +24,18 @@
 
 PyObject *idToKey(JSContext *cx, JS::HandleId id) {
   JS::RootedValue idv(cx, js::IdToValue(id));
-  JSString *idStr;
+  JS::RootedString idStr(cx);
   if (!id.isSymbol()) { // `JS::ToString` returns `nullptr` for JS symbols
     idStr = JS::ToString(cx, idv);
   } else {
     // TODO (Tom Tang): Revisit this once we have Symbol coercion support
     // FIXME (Tom Tang): key collision for symbols without a description string, or pure strings look like "Symbol(xxx)"
     idStr = JS_ValueToSource(cx, idv);
   }
+
   // We convert all types of property keys to string
-  return StrType(cx, idStr).getPyObject();
+  auto chars = JS_EncodeStringToUTF8(cx, idStr);
+  return PyUnicode_FromString(chars.get());
```

Comment:
Why are we encoding to utf8 and then making a Python String from that instead of making a Python String from the already existing latin1/utf16 encoding from the JSString?

Reply:
The string may be outside the range of the Basic Multilingual Plane, which must be encoded as USC4 string in Python. 
Also the property key string would be short so we could afford copying the backing store.